Languages/IronPython/IronPython/Runtime/Importer.cs

```diff
@@ -917,8 +917,19 @@ class PythonFileStreamContentProvider : StreamContentProvider {
                 try {
                     object handler = PythonCalls.Call(context, hook, dirname);
 
-                    if (handler != null) {
-                        return handler;
+                    if (handler != null)
+                    {
+                        if (handler is IImporterModule)
+                        {
+                            if ((handler as IImporterModule).State == ImporterModuleState.Ready)
```

Comment:
Use an `((IImporterModule)handler).State; you've already checked with `is`, no need to check again with `as`.